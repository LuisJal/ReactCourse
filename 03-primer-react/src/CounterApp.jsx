import React from "react";
import PropTypes from "prop-types";
import { useState } from "react";

export const CounterApp = ({ value }) => {
  const [counter, setCounter] = useState(value);

  //Sumar
  const handleAdd = () => setCounter((c) => c + 1);
  //Restar
  const handleSubtract = () => setCounter((c) => c - 1);
  //Reiniar
  const handleReset = () => setCounter((c) => (c = 0));

  return (
    <>
      <h1>CounterAppLuis</h1>
      <h2 data-testid="contador">{counter}</h2>
      <div className="botones">
        <button onClick={handleAdd}>+1</button>
        <button onClick={handleSubtract}>-1</button>
        <button onClick={handleReset}>Reset</button>
      </div>
    </>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
};
