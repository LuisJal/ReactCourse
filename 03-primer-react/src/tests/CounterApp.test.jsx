import { CounterApp } from "../CounterApp";
import { fireEvent, getByTestId, render, screen } from "@testing-library/react";

describe("Pruebas en CounterApp", () => {

    const initValue = 100;
  
  test("debe hacer match con el snapshot", () => {
    
    const { container } = render(<CounterApp value={initValue} />);

    expect(container).toMatchSnapshot();
  });

  test("debe mostrar el valor inicial de 100", () => {
    const {  getByTestId } = render(
        <CounterApp value={initValue} />
      );
      expect(screen.getByTestId("contador").innerHTML).toBe("100");

  });

  test("debe incrementar con el boton", () =>{
    render(<CounterApp value={initValue} />)
    fireEvent.click(screen.getByText('+1'))

    expect(screen.getByTestId("contador").innerHTML).toBe("101");
    

  });



  test("debe decrementar con el boton", () =>{
    render(<CounterApp value={initValue} />)
    fireEvent.click(screen.getByText('-1'))

    expect(screen.getByTestId("contador").innerHTML).toBe("99");
    

  });

  test("debe resetear con el boton", () =>{
    render(<CounterApp value={initValue} />)
    fireEvent.click(screen.getByText('Reset'))

    expect(screen.getByTestId("contador").innerHTML).toBe("0");
    

  });

  
});
