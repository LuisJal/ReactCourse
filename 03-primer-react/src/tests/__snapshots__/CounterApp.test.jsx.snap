// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en CounterApp debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    CounterAppLuis
  </h1>
  <h2
    data-testid="contador"
  >
    100
  </h2>
  <div
    class="botones"
  >
    <button>
      +1
    </button>
    <button>
      -1
    </button>
    <button>
      Reset
    </button>
  </div>
</div>
`;
